### PR TITLE
[ MB-14885 ] Convert middleware to store userID instead of the full user model in the request context

### PR DESCRIPTION
**Pull request by mmayo-truss**
## Summary

This PR changes middleware from storing the user model to storing the user id in the request context. It is meant to avoid the database call at this point in the middleware in response to an issue that arose in the on call channel. 

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Verify that move history events are still storing the information of the user and that information is displaying in the move history table.

### Screenshots
The move history table is functioning as expected. 
image

### CODE REVIEW

**Comment by mmayo-truss** [comment]
@Mergifyio rebase

File: pkg/handlers/authentication/client_cert.go [hunk 1]
```diff
@@ -60,13 +57,7 @@ func ClientCertMiddleware(appCtx appcontext.AppContext) func(next http.Handler)
 				return
 			}
 			ctx := SetClientCertInRequestContext(r, clientCert)
-			user, err := user.NewUserFetcher(query.NewQueryBuilder()).FetchUser(newAppCtx, []services.QueryFilter{query.NewQueryFilter("id", "=", clientCert.UserID)})
```
Comment (duncan-truss) [inline pkg/handlers/authentication/client_cert.go:63]: Did you also see this query failing in the logs or you just think it's best to remove it too?

File: pkg/services/audit/audit.go [hunk 2]
```diff
@@ -104,16 +103,6 @@ func CaptureAccountStatus(appCtx appcontext.AppContext, model interface{}, activ
 	return logItems, nil
 }
 
-func extractAuditUser(request *http.Request) []zap.Field {
-	var logItems []zap.Field
-	auditUser := audit.RetrieveAuditUserFromContext(request.Context())
-	logItems = append(logItems,
-		zap.String("audit_user_id", auditUser.ID.String()),
-		zap.String("audit_user_email", auditUser.LoginGovEmail),
```
Comment (duncan-truss) [inline pkg/services/audit/audit.go:112]: This is just logging related right, we don't rely on this for setting the email in the actual audit history?

**Review by duncan-truss — APPROVED** [review]

File: pkg/handlers/authentication/client_cert.go (hunk 1 above)
Comment (mmayo-truss) [reply pkg/handlers/authentication/client_cert.go:63]: I did not see it failing. But the only thing being done with the user was passing it to the request context.

**Review by afoon — APPROVED** [review]
lgtm! Did not see any regressions when view move history in frontend or database

**Pull request merged**